enterprise/backend/test/metabase_enterprise/advanced_config/file/api_keys_test.clj

```diff
@@ -31,125 +31,214 @@
-                                                               :group "all-users"}]}}]
-            (is (= :ok (config.file/initialize!)))
-            (testing "API keys should be created"
-              (is (api-key-exists? "Test API Key"))
-              (is (api-key-exists? "All Users API Key")))
-            (testing "API key users should be created"
-              (is (api-key-user-exists? "Test API Key"))
-              (is (api-key-user-exists? "All Users API Key"))))
-          (finally
-            (cleanup-config!))))
-
-      (testing "should fail if API keys have the same prefix"
-        (try
-          (write-config!
-           {:version 1
-            :config {:api-keys [{:name "First API Key"
-                                 :key "mb_sameprefix_123"
-                                 :creator "admin@test.com"
-                                 :group "admin"}]}})
-          (binding [config.file/*config* {:version 1
-                                          :config {:api-keys [{:name "First API Key"
-                                                               :key "mb_sameprefix123"
-                                                               :creator "admin@test.com"
-                                                               :group "admin"}
-                                                              {:name "Second API Key"
-                                                               :key "mb_sameprefix456"
-                                                               :creator "admin@test.com"
-                                                               :group "admin"}]}}]
-            (is (thrown-with-msg?
-                 clojure.lang.ExceptionInfo
-                 #"API key with prefix 'mb_same' already exists\. Keys must have unique prefixes\."
-                 (config.file/initialize!))))
-          (finally
-            (cleanup-config!))))
-
-      (testing "should fail if creator is not an admin"
-        (try
-          (mt/with-temp [:model/User _ {:email "regular@test.com"
-                                        :first_name "Regular"
-                                        :is_superuser false}]
-            (binding [config.file/*config* {:version 1
-                                            :config {:api-keys [{:name "Test API Key"
-                                                                 :key "mb_1testapikey123"
-                                                                 :creator "regular@test.com"
-                                                                 :group "admin"}]}}]
-              (is (thrown-with-msg?
-                   clojure.lang.ExceptionInfo
-                   #"User with email regular@test.com is not an admin"
-                   (config.file/initialize!)))))
-          (finally
-            (cleanup-config!))))
-
-      (testing "should fail if creator doesn't exist"
-        (try
+      (try
+        (write-config!
```

Comment:
I should have noticed this before, but what's the difference between `write-config!` with the map and binding `config.file/*config*` to the map? Why do you need both?

Reply:
We don't :/

Yeah write config creates a physical file while binding does in memory and takes precedence, so I'm keeping only the binding